libraries/extensions/telemetry/tracing/src/lib.rs

```diff
@@ -4,19 +4,53 @@
 //! able to serialize and deserialize context that has been sent via the middleware.
 
 use eyre::Context as EyreContext;
-use tracing_subscriber::{EnvFilter, Layer};
+use tracing::metadata::LevelFilter;
+use tracing_subscriber::{prelude::__tracing_subscriber_SubscriberExt, EnvFilter, Layer};
 
 #[cfg(feature = "telemetry")]
 pub mod telemetry;
 
-pub fn set_up_tracing() -> eyre::Result<()> {
-    use tracing_subscriber::prelude::__tracing_subscriber_SubscriberExt;
+pub fn set_up_tracing(name: &str) -> eyre::Result<()> {
+    // Filter log using `RUST_LOG`. More useful for CLI.
     let filter = EnvFilter::from_default_env();
-
     let stdout_log = tracing_subscriber::fmt::layer()
         .pretty()
         .with_filter(filter);
+
+    #[cfg(feature = "telemetry")]
+    let subscriber = {
+        use eyre::ContextCompat;
+        use opentelemetry::sdk::export::trace::stdout;
+        use tracing_subscriber::Registry;
+
+        let filter = EnvFilter::from_default_env();
+        let telemetry = match std::env::var_os("DORA_JAEGER_TRACING") {
+            Some(endpoint) => {
+                let endpoint = endpoint
+                    .to_str()
+                    .wrap_err("Could not parse env variable: DORA_JAEGER_TRACING")?;
+                let tracer = crate::telemetry::init_jaeger_tracing(name, endpoint)
+                    .wrap_err("Could not instantiate tracing")?;
+                tracing_opentelemetry::layer()
+                    .with_tracer(tracer)
+                    .with_filter(filter.add_directive(LevelFilter::TRACE.into()))
+            }
+            None => {
+                let tracer = stdout::new_pipeline().install_simple();
```

Comment:
Is this `stdout` trace in a human-readable format?

Reply:
So, I added the pretty print option in the `stdout::new_pipeline` from opentelemetry, which makes it better, 
From:
```bash
SpanData { span_context: SpanContext { trace_id: 0a6e102fea3d339013e23164c2394cb4, span_id: fb64a8354f00854f, trace_flags: TraceFlags(1), is_remote: false, trace_state: TraceState(None) }, parent_span_id: 0000000000000000, span_kind: Internal, name: "handle_connection_loop", start_time: SystemTime { tv_sec: 1679885600, tv_nsec: 94738878 }, end_time: SystemTime { tv_sec: 1679885600, tv_nsec: 739831257 }, attributes: EvictedHashMap { map: {Static("thread.id"): I64(2), Static("idle_ns"): I64(644847345), Static("queue_sizes"): String(Owned("{DataId(\"op/latency\"): 10}")), Static("code.namespace"): String(Static("dora_daemon::listener::tcp")), Static("code.filepath"): String(Static("binaries/daemon/src/listener/tcp.rs")), Static("busy_ns"): I64(243993), Static("code.lineno"): I64(44), Static("thread.name"): String(Owned("tokio-runtime-worker"))}, evict_list: [Static("idle_ns"), Static("busy_ns"), Static("queue_sizes"), Static("thread.name"), Static("thread.id"), Static("code.lineno"), Static("code.namespace"), Static("code.filepath")], max_len: 128, dropped_count: 0 }, events: EvictedQueue { queue: Some([Event { name: "channel disconnected: a264d253-4d9b-449f-ab44-d47343f42939/op_2", timestamp: SystemTime { tv_sec: 1679885600, tv_nsec: 739779000 }, attributes: [KeyValue { key: Static("level"), value: String(Static("DEBUG")) }, KeyValue { key: Static("target"), value: String(Static("dora_daemon::listener")) }, KeyValue { key: Static("code.filepath"), value: String(Static("binaries/daemon/src/listener/mod.rs")) }, KeyValue { key: Static("code.namespace"), value: String(Static("dora_daemon::listener")) }, KeyValue { key: Static("code.lineno"), value: I64(190) }], dropped_attributes_count: 0 }]), max_len: 128, dropped_count: 0 }, links: EvictedQueue { queue: None, max_len: 128, dropped_count: 0 }, status: Unset, resource: Resource { attrs: {Static("service.name"): String(Owned("unknown_service"))}, schema_url: None }, instrumentation_lib: InstrumentationLibrary { name: "opentelemetry", version: Some("0.18.0"), schema_url: None } }
```

To:
```bash
SpanData {
    span_context: SpanContext {
        trace_id: b42f990b8b761cbf4331b1b12393a168,
        span_id: 2c00b6ddc59e80ad,
        trace_flags: TraceFlags(
            1,
        ),
        is_remote: false,
        trace_state: TraceState(
            None,
        ),
    },
    parent_span_id: 0000000000000000,
    span_kind: Internal,
    name: "listener_loop",
    start_time: SystemTime {
        tv_sec: 1679885225,
        tv_nsec: 530300305,
    },
    end_time: SystemTime {
        tv_sec: 1679885226,
        tv_nsec: 974209562,
    },
    attributes: EvictedHashMap {
        map: {
            Static(
                "thread.id",
            ): I64(
                12,
            ),
            Static(
                "code.lineno",
            ): I64(
                18,
            ),
            Static(
                "thread.name",
            ): String(
                Owned(
                    "tokio-runtime-worker",
                ),
            ),
            Static(
                "queue_sizes",
            ): String(
                Owned(
                    "{DataId(\"op/latency\"): 10}",
                ),
            ),
            Static(
                "idle_ns",
            ): I64(
                244581990,
            ),
            Static(
                "code.filepath",
            ): String(
                Static(
                    "binaries/daemon/src/listener/tcp.rs",
                ),
            ),
            Static(
                "code.namespace",
            ): String(
                Static(
                    "dora_daemon::listener::tcp",
                ),
            ),
            Static(
                "busy_ns",
            ): I64(
                60210,
            ),
        },
        evict_list: [
            Static(
                "idle_ns",
            ),
            Static(
                "busy_ns",
            ),
            Static(
                "queue_sizes",
            ),
            Static(
                "thread.name",
            ),
            Static(
                "thread.id",
            ),
            Static(
                "code.lineno",
            ),
            Static(
                "code.namespace",
            ),
            Static(
                "code.filepath",
            ),
        ],
        max_len: 128,
        dropped_count: 0,
    },
    events: EvictedQueue {
        queue: None,
        max_len: 128,
        dropped_count: 0,
    },
    links: EvictedQueue {
        queue: None,
        max_len: 128,
        dropped_count: 0,
    },
    status: Unset,
    resource: Resource {
        attrs: {
            Static(
                "service.name",
            ): String(
                Owned(
                    "unknown_service",
                ),
            ),
        },
        schema_url: None,
    },
    instrumentation_lib: InstrumentationLibrary {
        name: "opentelemetry",
        version: Some(
            "0.18.0",
        ),
        schema_url: None,
    },
}
```

which is *better* but the coloring and the pretty format from tokio is not working. The log is quite verbose as well and might be hard to use as-it-is without Jaeger.

Reply:
Absolutely. I think it is the most natural use case.